native~/Runtime/src/UnityPrepareRendererResources.cpp

```diff
@@ -1289,21 +1475,91 @@ void* UnityPrepareRendererResources::prepareInMainThread(
                     static_cast<float>(texCoordIndexIt->second));
               }
             }
           }
+
+          const ExtensionKhrTextureTransform* pBaseColorTextureTransform =
+              pbr.baseColorTexture
+                  ? pbr.baseColorTexture
+                        ->getExtension<ExtensionKhrTextureTransform>()
+                  : nullptr;
+          if (pBaseColorTextureTransform) {
+            UnityEngine::Vector2 offset{
+                (float)pBaseColorTextureTransform->offset[0],
+                (float)pBaseColorTextureTransform->offset[1]};
+            UnityEngine::Vector2 scale{
+                (float)pBaseColorTextureTransform->scale[0],
+                (float)pBaseColorTextureTransform->scale[1]};
+            material.SetTextureOffset(
+                shaderProperties.getBaseColorTextureID(),
+                offset);
+            material.SetTextureScale(
+                shaderProperties.getBaseColorTextureID(),
+                scale);
+          }
+
+          const ExtensionKhrTextureTransform*
+              pMetallicRoughnessTextureTransform =
+                  pbr.metallicRoughnessTexture
+                      ? pbr.metallicRoughnessTexture
+                            ->getExtension<ExtensionKhrTextureTransform>()
+                      : nullptr;
+          if (pMetallicRoughnessTextureTransform) {
+            UnityEngine::Vector2 offset{
+                (float)pMetallicRoughnessTextureTransform->offset[0],
+                (float)pMetallicRoughnessTextureTransform->offset[1]};
+            UnityEngine::Vector2 scale{
+                (float)pMetallicRoughnessTextureTransform->scale[0],
+                (float)pMetallicRoughnessTextureTransform->scale[1]};
+            material.SetTextureOffset(
+                shaderProperties.getMetallicRoughnessTextureID(),
+                offset);
+            material.SetTextureScale(
+                shaderProperties.getMetallicRoughnessTextureID(),
+                scale);
+          }
+
+          if (pBaseColorTextureTransform ||
+              pMetallicRoughnessTextureTransform) {
+            UnityEngine::Vector4 rotationValues{0.0f, 1.0f, 0.0f, 1.0f};
+            if (pBaseColorTextureTransform) {
+              rotationValues.x =
+                  float(glm::sin(pBaseColorTextureTransform->rotation));
+              rotationValues.y =
+                  float(glm::cos(pBaseColorTextureTransform->rotation));
+            }
+            if (pMetallicRoughnessTextureTransform) {
+              rotationValues.z =
+                  float(glm::sin(pMetallicRoughnessTextureTransform->rotation));
+              rotationValues.w =
+                  float(glm::cos(pMetallicRoughnessTextureTransform->rotation));
+            }
+
+            material.SetVector(
+                shaderProperties.getBaseColorMetallicRoughnessRotationID(),
+                rotationValues);
+          }
+
+          // setGltfMaterialParameterValues(
```

Comment:
If I use the `setGltfMaterialParameterValues` function, which basically contains a copy of the code above it, Unity guaranteed crashes in the `CesiumShaderProperties` constructor because of a "heap corruption"... what am I doing so incredibly wrong? :sob:
![image](https://github.com/CesiumGS/cesium-unity/assets/32226860/0718e9b1-4e91-4730-957a-f69e9810e8e4)

(Of course, this doesn't happen in the `Debug` build...)